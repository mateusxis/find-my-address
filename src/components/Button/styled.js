import styled from 'styled-components';

export const SButton = styled.button`
  border: 0px;
  color: rgb(66, 66, 66);
  background-color: rgb(126, 194, 68);
  width: 80%;
  height: 62px;
  border-radius: 0.4rem;
  margin-top: 1rem;
  font-weight: bold;
  font-size: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  
  &::placeholder {
    color: #bdbdbd;
    text-transform: uppercase;
  }
  
  &:hover,
  &:focus {
    outline: none;
    box-shadow: 0.2rem 0.8rem 1.6rem #5e35b1;c
  }
  
  @media (min-width: 601px) {
    margin-left: 1.6rem;
    margin-top: 0;
    font-size: 1.2rem;
    line-height: 1.2rem;
    height: 78px;
  }

  @media (min-width: 961px) {
    font-size: 1.6rem;
    line-height: 1.6rem;
    height: 100px;
    width: 300px;
  }
`;
