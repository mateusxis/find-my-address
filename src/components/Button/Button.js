import { SButton } from './styled';
import Loading from '../Loading';

const Button = ({ children, onClick, loading = false }) => {
  return <SButton onClick={onClick}>{loading ? <Loading /> : children}</SButton>;
};

export default Button;
