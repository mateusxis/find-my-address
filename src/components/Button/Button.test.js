import renderer from 'react-test-renderer';

import Button from './Button';

jest.mock('../Loading', () => jest.fn(() => <></>));

describe('Button', () => {
  it('should render Button with props default values', () => {
    const tree = renderer.create(<Button onClick={() => jest.fn()}>Send</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Button with prop loading', () => {
    const tree = renderer
      .create(
        <Button onClick={() => jest.fn()} loading={true}>
          Send
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
