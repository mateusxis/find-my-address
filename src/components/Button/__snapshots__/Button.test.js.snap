// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button with prop loading 1`] = `
<button
  className="sc-bdvvtL dAXaRU"
  onClick={[Function]}
/>
`;

exports[`Button should render Button with props default values 1`] = `
<button
  className="sc-bdvvtL dAXaRU"
  onClick={[Function]}
>
  Send
</button>
`;
